include/aspect/particle/particle.h

```diff
@@ -21,119 +21,141 @@
 #ifndef __aspect__particle_particle_h
 #define __aspect__particle_particle_h
 
-#include <aspect/postprocess/interface.h>
-#include <aspect/simulator_access.h>
+#include <aspect/global.h>
+
+#include <deal.II/base/point.h>
+
+#include <boost/serialization/vector.hpp>
 
 namespace aspect
 {
   namespace Particle
   {
+    using namespace dealii;
+
+
+    /* Type definitions */
+
+#ifdef DEAL_II_WITH_64BIT_INDICES
     /**
-     * Typedef of cell level/index pair
+     * The type used for indices of tracers. While in
+     * sequential computations the 4 billion indices of 32-bit unsigned integers
+     * is plenty, parallel computations using hundreds of processes can overflow
+     * this number and we need a bigger index space. We here utilize the same
+     * build variable that controls the dof indices of deal.II because the number
+     * of degrees of freedom and the number of tracers are typically on the same
+     * order of magnitude.
+     *
+     * The data type always indicates an unsigned integer type.
      */
-    typedef std::pair<int, int> LevelInd;
+    typedef unsigned long long int particle_index;
```

Comment:
(As for location, I suggested `definitions.h` because you moved the other typedef there. I don't actually care very much whether you have it here or there, but it seems useful to keep them in the same place. This may actually be the better place.)